src/ILLink.RoslynAnalyzer/COMAnalyzer.cs

```diff
@@ -39,6 +40,13 @@ public override void Initialize (AnalysisContext context)
 					if (!targetMethod.HasAttribute (DllImportAttribute))
 						return;
 
+					if (operationContext.ContainingSymbol is ISymbol containingSymbol) {
+						if (containingSymbol.HasAttribute (RequiresUnreferencedCodeAttribute)
+							|| containingSymbol.ContainingType.HasAttribute (RequiresUnreferencedCodeAttribute)) {
+							return;
```

Comment:
The "RUC on type" only applies to the type itself, it does not propagate into nested types. So similar to https://github.com/dotnet/linker/pull/2400/files we need to only check the `ContainingType` if the symbol is not a type.

Ideally we would share this logic between these places, as it should always be the same.

Reply:
I see, I've updated the code. Should this functionality be moved to the ISymbol extensions to share it?

Tagging @tlakollo for input.

Reply:
I don't have a strong opinion for having some methods in the ISymbol extensions file, but given that is a copy of Roslyn's ISymbolExtensions with some modifications and 'Requires' is very linker focused thing, plus the fact that the desired behavior will be in the future to have the implementation shared between linker and analyzer I think it will be better to have a separate file.

Reply:
Now that Tlaka's change is in the main, I think we should move this common logic into some helper somewhere and use it from both places.